### PR TITLE
CASMINST-4176 retry skew check three times to reduce false positives

### PR DESCRIPTION
**Pull request by jacobsalmela**
Signed-off-by: Jacob Salmela <jacob.salmela@hpe.com>

## Summary and Scope

Wraps the existing check in a loop that tries three times.

## Issues and Related PRs

* Resolves CASMINST-4176

## Testing

- `redbull`
- `fanta`

### Test description:

```
ncn-m001:~/jsalmela # ./!$
./test
Checking clock skew...attempt 1 of 3...
Epoch seconds by node (allowing 1 seconds of drift):
ncn-s001: 1646746048
ncn-s002: 1646746048
ncn-s003: 1646746048
ncn-s004: 1646746048
ncn-m001: 1646746048
ncn-m002: 1646746048
ncn-m003: 1646746048
ncn-w001: 1646746048
ncn-w002: 1646746049
ncn-w004: 1646746049
ncn-w003: 1646746049

Checking clock skew...attempt 2 of 3...
Epoch seconds by node (allowing 1 seconds of drift):
ncn-s001: 1646746059
ncn-s003: 1646746059
ncn-s002: 1646746059
ncn-s004: 1646746059
ncn-m001: 1646746059
ncn-m002: 1646746059
ncn-m003: 1646746059
ncn-w001: 1646746059
ncn-w002: 1646746060
ncn-w004: 1646746060
ncn-w003: 1646746060

Checking clock skew...attempt 3 of 3...
Epoch seconds by node (allowing 1 seconds of drift):
ncn-s002: 1646746070
ncn-s001: 1646746070
ncn-s003: 1646746070
ncn-s004: 1646746070
ncn-m001: 1646746070
ncn-m002: 1646746070
ncn-m003: 1646746070
ncn-w001: 1646746070
ncn-w002: 1646746071
ncn-w004: 1646746071
ncn-w003: 1646746071

ncn-m001:~/jsalmela # echo $?
0
ncn-m001:~/jsalmela #
```

## Risks and Mitigations

Low.  This just loops existing code three times with a small delay.

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: goss-testing/scripts/check_clock_skew.sh [hunk 1]
```diff
@@ -49,6 +49,11 @@ for node in $nodes; do
   pdsh_node_args="$pdsh_node_args -w $node"
 done
 
+for try in {1..3}; do
```
Comment (dzou-hpe) [inline goss-testing/scripts/check_clock_skew.sh:52]: should break out of the loop when no clock skew is detected?

**Review by dzou-hpe — APPROVED** [review]

File: goss-testing/scripts/check_clock_skew.sh (hunk 1 above)
Comment (jacobsalmela) [reply goss-testing/scripts/check_clock_skew.sh:75]: We could.  I was just trying to keep the changes short to address the intermittent issues seen with the test.  I don't think it'e entirely necessary despite it adding 30 seconds to the test.

Comment (heemstra) [reply goss-testing/scripts/check_clock_skew.sh:75]: I think we should either break out on success or shorten the sleep to like 2s.  Or was 10 chosen based on real-life observations?

Comment (jacobsalmela) [reply goss-testing/scripts/check_clock_skew.sh:75]: Just an arbitrary value I chose.  I'll reduce to 2

Comment (jacobsalmela) [reply goss-testing/scripts/check_clock_skew.sh:75]: Fixed to `sleep 2`

**Comment by jacobsalmela** [comment]
wrong branch.  See https://github.com/Cray-HPE/csm-testing/pull/228

**Pull request closed without merging**